packages/server-boilerplate/src/orchestrator/api/ApiBuilder.ts

```diff
@@ -62,6 +63,17 @@ export class ApiBuilder {
     this.attachVerifyLogin = emptyMiddleware;
     this.verifyAuthMiddleware = emptyMiddleware; // Do nothing by default
     this.attachAccessPolicy = buildAttachAccessPolicy(new AccessPolicyBuilder(this.models));
+
+    // Dynamically set trusted proxy so that we can trust the IP address of the client
+    publicIp
+      .v4()
+      .then(publicIp => {
+        this.app.set('trust proxy', ['loopback', process.env.AWS_TRUSTED_PROXY_IP, publicIp]);
```

Comment:
My only concern with this is that due to AWS elastic IPs, I think it's possible for the server's IP address to change while server is running. There's a thread about it here: https://beyondessential.slack.com/archives/CT0KTFSHZ/p1717105831450469

Would it be crazy if we set this on an interval?

Reply:
hmmm it is a bit funky but apart from it feeling bad I don't actually see any concrete reason not to

Reply:
@edmofro Can I get your 2 cents on this one before I implement it, since it is a bit of a new pattern?

Reply:
Yeah agree it's a little funky but I think we need to do it! So long as there's commenting around it I think it's fine.